histFactory/src/createPlotter.cpp

```diff
@@ -364,25 +372,52 @@ bool get_plots_files(const std::string& python_file, std::vector<Plot>& plots, s
             std::cerr << "The 'before_loop' variable is not a string" << std::endl;
             return false;
         }
-        userCode.before_loop = PyString_AsString(py_before_loop);
+        extra.userCode.before_loop = PyString_AsString(py_before_loop);
     }
     PyObject* py_in_loop = PyDict_GetItemString(global_dict, "code_in_loop");
     if (py_in_loop) {
         if (! PyString_Check(py_in_loop)) {
             std::cerr << "The 'in_loop' variable is not a string" << std::endl;
             return false;
         }
-        userCode.in_loop = PyString_AsString(py_in_loop);
+        extra.userCode.in_loop = PyString_AsString(py_in_loop);
     }
     PyObject* py_after_loop = PyDict_GetItemString(global_dict, "code_after_loop");
     if (py_after_loop) {
         if (! PyString_Check(py_after_loop)) {
             std::cerr << "The 'after_loop' variable is not a string" << std::endl;
             return false;
         }
-        userCode.after_loop = PyString_AsString(py_after_loop);
+        extra.userCode.after_loop = PyString_AsString(py_after_loop);
+    }
+
+    // Retrieve dict of sample weights
+    // The key can be set in the sample json to indicate which sample weight to use
+    // The value is any valid C++ code, which will be added in the global `getSampleWeight`
+    // function.
+    PyObject* py_sample_weights = PyDict_GetItemString(global_dict, "sample_weights");
+    if (py_sample_weights) {
+        if (! PyDict_Check(py_sample_weights)) {
+            std::cerr << "The 'sample_weights' variable is not a dictionnary" << std::endl;
+            return false;
+        }
+
+        PyObject* key, *value;
+        Py_ssize_t pos = 0;
+        while (PyDict_Next(py_sample_weights, &pos, &key, &value)) {
+            if (! PyString_Check(key)) {
+                std::cerr << "Keys of 'sample_weights' dict must be strings" << std::endl;
+            }
+
+            if (! PyString_Check(value)) {
+                std::cerr << "Values of 'sample_weights' dict must be strings" << std::endl;
```

Comment:
Maybe add a `continue` here and above? Or do you prefer `PyString_AsString` to throw an exception?